### ISSUE TITLE
Specifying YAML files via the command line

### ISSUE THREAD

**Issue by connorp** [quarto-dev/quarto-cli#2971]
### Bug description

I am attempting to pass a YAML file of execution parameters via the command line option `--execute-params`: 

`test.qmd` file: 
```
---
title: test
date: now
format: pdf
---

Test document.
```

`test.yml` file:
```
date-format: MMM DD, YYYY HH:mm:ss z
```

Command line: 
```
quarto render test.qmd --execute-params test.yml
```

Using this command, I would expect the date to be formatted in the manner specified in the YAML file, but the rendered file uses the default long format. Am I using the command line argument incorrectly, or is there an issue here? 

Running macOS version 12.6, and Quarto version 1.1.179. 

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by connorp** [comment]
Nevermind, I figured out `--execute-params` is the wrong argument to use. Still unsure how to pass a YAML file of metadata on the command line, but that's a separate question.

**Comment by mervebduman** [comment]
It was working just fine for me when I have

*test.qmd:*
```
---
title: "Test Quarto Report"
---
```

*_quarto.yml:*
```
project:
  title: ""
```

*params.yml:*
```
project:
  params:
    data: "~/test/test.csv"
```

and I execute by: 
```shell
quarto render --execute-params params.yml
```

Suddenly this stopped working, and I have no idea why. Maybe it works for you.

**Comment by mervebduman** [comment]
Hey, thanks for reaching out!

[Here](https://github.com/mervebduman/quarto-params-test) is the repository.

The problem is that I don't want to define *params* in **.qmd** file.

I have the **data** in `params.yml` and right now my test.qmd looks like this:
````R
---
title: "Test Quarto Report"
---

```{r}
read.csv(params$data)
```
````

and I get the error below when I run `quarto render --execute-params params.yml`:

```shell
Quitting from lines 6-7 (test.qmd) 
Fehler in read.table(file = file, header = header, sep = sep, quote = quote,  : 
  Objekt 'params' nicht gefunden
Ruft auf: .main ... eval_with_user_handlers -> eval -> eval -> read.csv -> read.table

Ausführung angehalten
```

But if I add the `params:  data:""` lines like you do, it works perfectly fine:

````R
---
title: "Test Quarto Report"
params:
  data: ""
---
````

---
Long story short, I am trying to find a way to pass external params file to the .qmd file without having that variables in the document. 

Thank you for your interest, cheers!

**Comment by dragonstyle** [comment]
We require that you have those declared in the front matter as a way to ensure you could render the document while you're developing it. They also act as a default value should the params not be provided. Is there a reason why defining the `params` isn't going to work for you?

**Comment by mervebduman** [comment]
I am developing a report tool that requires to get data dynamically, and has modularised components. Since the final report always will be different, I would like to keep params in each components' _metadata.yml or in root folder's params.yml. I must have a params file always - so I don't need to have default values in the .qmd file.
